pkg/controllers/manifestwork_test.go

```diff
@@ -1641,3 +1642,43 @@ func TestValidateSecurityConstraints(t *testing.T) {
 	assert.Nil(t, err, "is nil when validating HypershiftDeploymentReconciler security constraints")
 	assert.True(t, passed, "when validating HostingCluster needs to be a ManagedCluster that is a member of a ManagedClusterSet")
 }
+
+func TestDeleteManifestworkWaitCleanUp(t *testing.T) {
+
+	client := initClient()
+	ctx := context.Background()
+	hdr := &HypershiftDeploymentReconciler{
+		Client:                  client,
+		Log:                     ctrl.Log.WithName("tester"),
+		ValidateClusterSecurity: false,
+	}
+
+	testHD := getHDforManifestWork()
+	testHD.Spec.HostingCluster = "local-cluster"
+
+	client.Create(ctx, testHD)
+	defer client.Delete(ctx, testHD)
+
+	mw, _ := scaffoldManifestwork(testHD)
+	client.Create(ctx, mw)
+	defer client.Delete(ctx, mw)
+
+	rqst, err := hdr.deleteManifestworkWaitCleanUp(ctx, testHD)
+	assert.Nil(t, err, "is nil when deleteManifestWorkWaitCleanUp is successful")
+	assert.EqualValues(t, ctrl.Result{RequeueAfter: 1 * time.Second, Requeue: true}, rqst, "request requeue should be 1s")
+
+	err = client.Get(ctx, types.NamespacedName{Name: mw.Name, Namespace: mw.Namespace}, mw)
+	mw.Status.Conditions = []metav1.Condition{
+		metav1.Condition{
+			Type:               string(workv1.WorkAvailable),
+			ObservedGeneration: mw.Generation,
+			Status:             metav1.ConditionTrue,
+		},
+	}
+	err = client.Update(ctx, mw)
+	assert.Nil(t, err, "is nil when condition is added")
+
+	rqst, err = hdr.deleteManifestworkWaitCleanUp(ctx, testHD)
+	assert.Nil(t, err, "is nil when deleteManifestWorkWaitCleanUp is successful")
+	assert.EqualValues(t, ctrl.Result{RequeueAfter: 20 * time.Second, Requeue: true}, rqst, "request requeue should be 20s")
```

Comment:
Probably better to also check the `testHD` status condition because the function modified it. Don't need to do a GET it's suppose to be updated from the param.

Reply:
I don't think there is any condition ont he testHD. The condition that is in question is on the ManifestWork, and we're not running a reconcile loop of that controller here.

The best I can do is check that at the end of the reconcile look, the ManifestWork Spec is set to the correct selectivelyOrphan value. Then in the 2nd part of the test.

1. So the test flow is, first reconcile on TestHD, we set the correct spec on the ManifestWork, but the manifestWork does not signal that the condition has been applied, so we early exist (expected behaviour).
2. We set the condition, verifying the spec change has been made, and run the reconcile on testHD again.  This time we detect that it has been reconciled, and we remove the manifest work.  

I decided to add a test, after the first run to make sure ManifestWork is still there, and after to make sure ManifestWork is NOT there.

Reply:
Inside `deleteManifestworkWaitCleanUp` function right before the return of 20 seconds there is: 

```
	//caller will execute the status update
	setStatusCondition(hyd, hypdeployment.WorkConfigured, metav1.ConditionTrue, "Removing HypershiftDeployment's manifestwork and related resources", hypdeployment.RemovingReason)
```

This sets in memory the HD status condition.

Reply:
Done in `https://github.com/stolostron/hypershift-deployment-controller/pull/123/files`